src/synapse/azext_synapse/_params.py

```diff
@@ -3,159 +3,204 @@
+        c.argument('default_spark_log_folder', arg_group='Default Folder', help='The default Spark log folder.')
 
         # Component Version
-        c.argument('spark_version', arg_group='Component Version', help='The supported spark version is 2.4 now.')
+        c.argument('spark_version', arg_group='Component Version', help='The supported Spark version is 2.4 now.')
 
-        c.argument('force', help='The flag of force operation.')
         c.argument('tags', arg_type=tags_type)
 
     with self.argument_context('synapse spark pool update') as c:
         c.argument('tags', arg_type=tags_type)
+        # Node
+        c.argument('node_count', arg_group='Node', help='The number of node.')
+        c.argument('node_size_family', arg_group='Node', help='The node size family.')
+
+        c.argument('node_size', arg_group='Node', arg_type=get_enum_type(['Small', 'Medium', 'Large']),
+                   help='The node size.')
+        # AutoScale
+        c.argument('enable_auto_scale', arg_type=get_three_state_flag(), arg_group='AutoScale',
+                   help='The flag of enabling auto scale.')
+        c.argument('max_node_count', arg_group='AutoScale', help='The max node count.')
+        c.argument('min_node_count', arg_group='AutoScale', help='The min node count.')
+
+        # AutoPause
+        c.argument('enable_auto_pause', arg_type=get_three_state_flag(), arg_group='AutoPause',
+                   help='The flag of enabling auto pause.')
+        c.argument('delay_in_minutes', arg_group='AutoPause', help='The delay time.')
+
+        # Environment Configuration
+        c.argument('library_requirements_file', arg_group='Environment Configuration',
+                   help='The library requirements file.')
+        c.argument('force', arg_type=get_three_state_flag(), help='The flag of force operation.')
 
     # synapse sql pool
     with self.argument_context('synapse sql pool') as c:
-        c.argument('workspace_name', help='The name of the workspace.')
+        c.argument('workspace_name', id_part='name', help='The workspace name.')
 
-    for scope in ['show', 'create', 'delete', 'pause', 'resume']:
+    with self.argument_context('synapse sql pool list') as c:
+        c.argument('workspace_name', id_part=None, help='The workspace name.')
+
+    for scope in ['show', 'create', 'delete', 'update', 'pause', 'resume']:
         with self.argument_context('synapse sql pool ' + scope) as c:
-            c.argument('sql_pool_name', arg_type=name_type, help='The sql pool name.')
+            c.argument('sql_pool_name', arg_type=name_type, id_part='child_name_1', help='The SQL pool name.')
 
     with self.argument_context('synapse sql pool create') as c:
         c.argument('max_size_bytes', help='The max size bytes.')
-        c.argument('sku_name', help='The sku name.')
+        c.argument('sku_name', options_list=['--performance-level'], help='The performance level.')

```

Comment:
>sku_name [](start = 20, length = 8)

why don't you change "sku_name" too?

Reply:
fixed.

---
In reply to: [424951128](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r424951128) [](ancestors = 424951128)